libs/base/string_utils.cpp

```diff
@@ -227,15 +227,15 @@ void AsciiToUpper(std::string & s)
 
 void Trim(std::string & s)
 {
-  boost::trim_if(s, ::isspace);
+  boost::trim_if(s, [](unsigned char c) { return std::isspace(c); });
```

Comment:
This is super interesting bug. For "à" letters at the end, the UTF-8 character consisting of two bytes is interpreted as one (last) byte, and this negative byte is converted to the unsigned char code 160 (non-breaking space). Then this "space" character is removed from the UTF-8 string on Mac OS. This breaks the validity of UTF-8 string. And in turn, it leads to the indefinite loop of the unsafe UTF-8 iteration over a corrupted UTF-8 character.

It reproduces in the test and in the debugger. But the funny part is that it _does not_ reproduce in a separate Trim test :)

Reply:
Mac OS X properly supports unicode spaces passed as int :)